### PR TITLE
Use useEmberModule from modules plugin to deprecate global Ember

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Related: 
 - emberjs/ember.js#19357
 - https://github.com/ember-cli/babel-plugin-ember-modules-api-polyfill/pull/175

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
I don't know how to test this, since we are `require`ing packages in the ember-plugins file

File: lib/ember-plugins.js [hunk 1]
```diff
@@ -80,6 +84,16 @@ function _shouldIgnoreJQuery(appRoot) {
...
 }
 
+function _shouldDeprecateGlobalEmber(appRoot) {
+  let packagePath = resolvePackagePath("ember-source", appRoot);
+  if (packagePath === null) {
+    return false;
+  }
+
+  let pkg = require(packagePath);
+  return pkg && semver.gte(pkg.version, "3.26.0-beta.1");
```
Comment (rwjblue) [inline lib/ember-plugins.js:94]:
```suggestion
  return pkg && semver.gte(pkg.version, "3.27.0-alpha.1");
```

**Review by rwjblue — CHANGES_REQUESTED** [review]
We can test with `fixturify-project`

**Comment by patocallaghan** [comment]
👋 @rwjblue I spoke to @NullVoxPopuli and I offered to get this PR over the line for him. I just have a question.

I see we've already updated `ember-plugins.js` to use `useEmberModule` but do we also need to add `useEmberModule` to `babel-options-util.js` too? What's the difference between the two files/usages?

https://github.com/babel/ember-cli-babel/blob/20924faa529cf4f9c6f7cb8fa2c89d968184b620/lib/babel-options-util.js#L132

**Comment by patocallaghan** [comment]
@rwjblue @NullVoxPopuli I've added a test to `node-tests/addon-test.js` which was set up already to use `fixturify-project`. It seems though these tests don't exercise the logic in `lib/ember-plugins.js` so I had to add the `useEmberModule` logic to `lib/babel-options-util.js` too.

**Comment by patocallaghan** [comment]
hmm, in the `floating dependencies` CI tests, i.e. `yarn install --no-lockfile`, the ember module gets transpiled to `_ember2` rather than `_ember`. I'm able to reproduce this locally. 

image

Any idea how the name of the AMD module is determined? why would it appear as `_ember2` in this scenario?

**Comment by rwjblue** [comment]
@patocallaghan - I think we should bump the babel-plugin-* that we have as dependencies, those failures are due to an update in one of those and bumping + fixing the fixtures in another PR should unblock.

**Comment by patocallaghan** [comment]
I've put together a PR in https://github.com/babel/ember-cli-babel/pull/385 which upgrades some deps. It mimics what we see with the "floating deps" in CI so it will resolve our issue

**Comment by rwjblue** [comment]
Hmm, I expected that to fail. Perhaps the issue isn't with one of those plugins?

**Comment by patocallaghan** [comment]
Ah no it is. Once I rebase this PR against that branch it starts failing. I've updated locally with _ember2 and it works fine

**Comment by patocallaghan** [comment]
I should say nothing else fails though

**Comment by patocallaghan** [comment]
@rwjblue I've rebased this against master to pick up the package upgrades. CI looks happy now 🎉

**Review by pzuraq — APPROVED** [review]

**Comment by patocallaghan** [comment]
@rwjblue any chance this could be merged and released please? I need this and the package upgrade PR in some other repo PRs

**Comment by rwjblue** [comment]
Closing in favor of #387

**Pull request closed without merging**